docs/api-docs-course/api-components.mdx

```diff
@@ -0,0 +1,644 @@
+
+
+
+![](https://paper-attachments.dropboxusercontent.com/s_D3923F54D30A795C87FAE705184D0CCF442B39BB28D6FFF1E1430BD3E9FB0903_1724205879784_image.png)
+
+
+*This image shows the Base URL of our Sample Petstore API*
+
+
+
+
+## AUTHORIZATION:
+
+For security reasons, not everyone is allowed to access certain areas in an API especially ones that contain sensitive information. API Data is protected by adding an Authentication Layer which is normally called **Auth**. This Authentication Layer then **verifies** that the person trying to make a request has permission **using Authorization.** 
+
+Authentication and Authorization work hand in hand to ensure API security but play unique roles. 
+Let’s say you want to make an API request to get data on dogs available in our Petstore:
+
+First, our Petstore API is protected since it contains sensitive information. You have to ask for the password access, that is Authentication. The authentication process here can be as simple as issuing you a unique username and password or using third-party tokens like Oauth, JWT etc. 
+
+Once you have gotten the password nothing happens except you key in the password into the system. Now Authorization steps in here to make sure that whatever credentials you have presented are approved.
+
+When that is done, you are in. 
+
+
+## **Types of Authentication** 
+
+
+1. **OAuth:** OAuth involves using a third-party provider to grant access so that you don't have to share your credentials with the API provider. An example is using your Google account to sign up for different platforms. This is a typical example of an OAuth authentication and how it works. 
+
+
+2. **JWT Bearer:** This uses JSON Web Tokens (JWTs), where the server issues a signed JWT to the client, and the client includes this token in the Authorization header of requests; the server then verifies the token's signature to ensure its validity and authenticity.
+****
+
+3. **API key:** An API key is a unique identifier issued by the API providers themselves. When making a request to a server that uses an API key, you have to include the API key as a parameter in the request body. 
+
+
+4. **Basic Oauth:** This is a simple authentication method that allows you to send HTTP requests by providing a username and password in plain text. 
+****
+
+
+**API Authentication in our** **Sample Petstore API**
+
+```json showLineNumber
+      "components": {
+        "securitySchemes": {
+          "ApiKeyAuth": {
+            "type": "apiKey",
+            "in": "header",
+            "name": "X-API-Key"
+          }
+        },
+    .
+    .
+    .
+    .
+    "security": [
+        {
+          "ApiKeyAuth": []
+        }
+      ],
+   ``` 
+
+Let’s understand the situation going on above. This is a snippet of how Authorization is used in an API document. 
+
+In the component `**securitySchemes**`, the type of authentication is defined as `**apiKey**`. This means that the API requires an API key for security.
```

Comment:
```suggestion
In the component `securitySchemes`, the type of authentication is defined as `apiKey`. This means that the API requires an API key for security.
```

Reply:
Remove all the astericks from the inline code blocks in the sentences below just like I did here